script/aria.js

```diff
@@ -7,746 +20,395 @@
-                '<dt><a href="#' +
-                item.title +
-                '" class="' +
-                item.is +
-                '-reference">' +
-                item.name +
-                "</a></dt>\n";
-            propIndex += "<dd>" + item.desc + "</dd>\n";
-        }
-        var node = document.getElementById("index_state_prop");
-        var parentNode = node.parentNode;
-        var l = document.createElement("dl");
-        l.id = "index_state_prop";
-        l.className = "compact";
-        l.innerHTML = propIndex;
-        parentNode.replaceChild(l, node);
-
-        var globalSPIndex = "";
-        sortedList = globalSP.sort(function (a, b) {
-            return a.name < b.name ? -1 : a.name > b.name ? 1 : 0;
-        });
-        for (i = 0; i < sortedList.length; i++) {
-            var lItem = sortedList[i];
-            globalSPIndex += "<li>";
-            if (lItem.is === "state") {
-                globalSPIndex +=
-                    "<sref " +
-                    (lItem.prohibited ? "data-prohibited " : "") +
-                    (lItem.deprecated ? "data-deprecated " : "") +
-                    'title="' +
-                    lItem.name +
-                    '">' +
-                    lItem.name +
-                    " (state)</sref>";
-            } else {
-                globalSPIndex +=
-                    "<pref " +
-                    (lItem.prohibited ? "data-prohibited " : "") +
-                    (lItem.deprecated ? "data-deprecated " : "") +
-                    ">" +
-                    lItem.name +
-                    "</pref>";
-            }
-            if (lItem.prohibited) {
-                globalSPIndex += " (Except where prohibited)";
-            }
-            if (lItem.deprecated) {
-                globalSPIndex += " (Global use deprecated in ARIA 1.2)";
-            }
-            globalSPIndex += "</li>\n";
-        }
-        parentNode = document.querySelector("#global_states");
-        if (parentNode) {
-            node = parentNode.querySelector(".placeholder");
-            if (node) {
-                l = document.createElement("ul");
-                l.innerHTML = globalSPIndex;
-                parentNode.replaceChild(l, node);
-            }
-        }
-        // there is only one role that uses the global properties
-        parentNode = document.querySelector(
-            "#roletype td.role-properties span.placeholder"
-        );
-        if (parentNode) {
-            node = parentNode.parentNode;
-            if (
-                (parentNode.textContent || parentNode.innerText) ===
-                "Placeholder for global states and properties"
-            ) {
-                l = document.createElement("ul");
-                l.innerHTML = globalSPIndex;
-                node.replaceChild(l, parentNode);
-            }
-        }
+/**
+ *
+ * @param {HTMLElement} item - rdef element
+ */
+const rewriteRdef = function (item) {
```

Comment:
at least if this is functionally similar to `generateHTMLStatesAndProperties`, it should have a similar name?